cmd/admin-handlers.go

```diff
@@ -1768,3 +1772,128 @@ func (a adminAPIHandlers) KMSKeyStatusHandler(w http.ResponseWriter, r *http.Req
 	}
 	writeSuccessResponseJSON(w, resp)
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+
+	switch madmin.InfoType(vars[madmin.INFO]) {
+	case madmin.ETCD:
+		errorString := ""
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoints := globalEtcdClient.Endpoints()
+		var etcdInfos []madmin.ETCDInfo
+		for _, endPoint := range etcdEndPoints {
+
+			if err := checkConnection(endPoint); err != nil {
+				errorString = errEtcdUnreachable.Error()
+			}
+
+			etcdInfo := madmin.ETCDInfo{
```

Comment:
As discussed, we could have one etcdInfo resp struct with `ETCDEndpoints` and `Errors` as slices

Reply:
done.